x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1181,6 +1186,9 @@ public int populateAndRead(
             final RangeAvailableHandler reader,
             final RangeMissingHandler writer
         ) throws Exception {
+            if (closed.get()) {
+                throwAlreadyClosed("Blob cache has been closed");
+            }
```

Comment:
Should we check that the shared bytes are closed in other APIs of SharedBlobCacheService as well and throwAlreadyClosed? I'm thinking tryRead or maybeFetchRegion/Range etc.

What I'm asking is whether we'd like to convert all possible ClosedChannelException to AlreadyClosedExceptions, rather than just in one API?